subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -122,76 +122,94 @@ func (h *HivedAlgorithm) Schedule(pod *core.Pod, suggestedNodes []string) intern
-			}
 			for podIndex := int32(0); podIndex < int32(len(gms.PodPlacements)); podIndex++ {
-				newGroup.physicalGpuPlacement[gpuNumber][podIndex] = make(
-					CellList, len(gms.PodPlacements[podIndex].PhysicalGpuIndices))
-				if hasVirtualPlacement {
-					newGroup.virtualGpuPlacement[gpuNumber][podIndex] = make(
-						CellList, len(gms.PodPlacements[podIndex].PhysicalGpuIndices))
-				}
 				node := gms.PodPlacements[podIndex].PhysicalNode
 				for gpuIndex := int32(0); gpuIndex < int32(
 					len(gms.PodPlacements[podIndex].PhysicalGpuIndices)); gpuIndex++ {
 					physicalGpuIndex := gms.PodPlacements[podIndex].PhysicalGpuIndices[gpuIndex]
-					pGpu := h.findPhysicalGpu(chain, node, physicalGpuIndex)
-					if pGpu == nil {
-						panic(fmt.Sprintf("[%v]: physical GPU cell not found when adding pod: node %v, GPU index %v",
-							internal.Key(pod), node, physicalGpuIndex))
-					}
-					newGroup.physicalGpuPlacement[gpuNumber][podIndex][gpuIndex] = pGpu
-
-					var vGpu *VirtualCell
-					if hasVirtualPlacement {
-						virtualCellIndex := gms.PodPlacements[podIndex].VirtualCellIndices[gpuIndex]
-						if virtualCellIndex >= 0 {
-							vGpu = h.findVirtualGpu(s.VirtualCluster, chain, s.ReservationId, virtualCellIndex)
-							if vGpu == nil {
-								panic(fmt.Sprintf("[%v]: virtual GPU cell not found when adding pod: virtual cell index %v",
-									internal.Key(pod), virtualCellIndex))
+					if pGpu := h.findPhysicalGpu(chain, node, physicalGpuIndex); pGpu == nil {
+						klog.Infof(
+							"[%v]: cannot find GPU %v on node %v due to reconfiguration: deleted from the spec",
+							internal.Key(pod), physicalGpuIndex, node)
+						klog.Infof("[%v]: pod ignored", internal.Key(pod))
+						break
+					} else {
+						newGroup.physicalGpuPlacement[gpuNumber][podIndex][gpuIndex] = pGpu
+						var vGpu *VirtualCell
+						if hasVirtualPlacement && !downgrade {
+							preassignedLevel := CellLevel(gms.PodPlacements[podIndex].PreassignedCellLevels[gpuIndex])
+							if preassignedLevel >= 0 {
+								var message string
+								if vcs := h.vcSchedulers[s.VirtualCluster]; vcs == nil {
+									message = fmt.Sprintf("VC %v has been deleted", s.VirtualCluster)
+								} else {
+									var vccl ChainCellList
+									var str string
+									if s.ReservationId != "" {
+										vccl = vcs.getReservedCellList()[s.ReservationId]
+										str = string(s.ReservationId)
+									} else {
+										vccl = vcs.getNonReservedCellList()[CellChain(info.CellChain)]
+										str = info.CellChain
+									}
+									if vccl == nil {
+										message = fmt.Sprintf("VC %v no longer has cells for %v", s.VirtualCluster, str)
+									} else {
+										vGpu, message = mapNonPreassignedCellToVirtual(pGpu, vccl, preassignedLevel)
+									}
+								}
+								if vGpu == nil {
+									klog.Infof(
+										"[%v]: cannot find virtual cell due to reconfiguration: %v",
```

Comment:
>reconfiguration [](start = 49, length = 15)

reconfiguration

Reply:
I don't understand..

---
In reply to: [339988743](https://github.com/microsoft/pai/pull/3768#discussion_r339988743) [](ancestors = 339988743)